src/App.tsx

```diff
@@ -32,15 +34,30 @@ export const App: React.FC = () => {
   }, []);
 
   const countActiveTodos = todos.reduce((accum, todo) => {
-    let countTodos = 0;
-
-    if (!todo.completed) {
-      countTodos += 1;
-    }
-
-    return accum + countTodos;
+    return !todo.completed ? accum + 1 : accum;
   }, 0);
 
+  const handleDeleteTodo = (id: number) => {
+    setIsLoading(true);
+    setDeleteTodoId(currentIds => [...currentIds, id]);
+
+    deleteTodo(id)
+      .then(() => {
+        setDeleteTodoId(currentIds =>
+          currentIds.filter(currId => currId !== id),
+        );
+
+        setTodos(currentTodos =>
+          currentTodos.filter(currentTodo => currentTodo.id !== id),
+        );
+        setIsLoading(false);
+      })
```

Comment:
You can simplify logic here with batch updates
deleteTodo(id)
  .then(() => {
    setTodos(currentTodos => {
      const newTodos = currentTodos.filter(todo => todo.id !== id);
      setDeleteTodoId(currentIds => currentIds.filter(currId => currId !== id));
      return newTodos;
    });
    setIsLoading(false);
  })